### PR TITLE
[Scheduling Sample] View Slot details, and update Encounter type mapping

### LINKED ISSUE #4937: [Scheduling Sample] View Slot details
- Clicking a Slot in the calendar should open a modal to show Slot details
  - Time 
  - Status
  - ~Appointments~
  - ~Location~

### PR DESCRIPTION
**Pull request by pamella**
## Description

- Resolves #4937 
- Add a Slot details modal with support for edit and delete actions
- ~Update the appointment and encounter creation flow:~
  - ~Map `Appointment.appointmentType` to `Encounter.type`~
  - ~Map `Appointment.serviceType` to `Encounter.serviceType`~

## Steps to test

- [ ] Build the bots: `npm run build:bots`
- [ ] Upload core data
- [ ] Upload example bots
- [ ] Check if all flows are working as expected
   - [ ] Create slot
   - [ ] Edit slot
   - [ ] Delete slot
   - [ ] Create appointment
   - [ ] Create encounter

[Screencast from 19-08-2024 12:35:45.webm](https://github.com/user-attachments/assets/4fc5b266-6b04-4901-8c88-7e72d1c4f265)

[Screencast from 19-08-2024 12:36:14.webm](https://github.com/user-attachments/assets/16833509-c27a-4fa8-80b7-7f7566718024)

[Screencast from 19-08-2024 12:37:23.webm](https://github.com/user-attachments/assets/c253b344-a280-41a3-a09d-81148ea57a37)

[Screencast from 19-08-2024 12:39:34.webm](https://github.com/user-attachments/assets/1b89b14f-0d0c-4f02-9077-ce38b75064ce)

### CODE REVIEW

File: examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx [hunk 1]
```diff
@@ -89,6 +89,7 @@ export function AppointmentActions(props: AppointmentActionsProps): JSX.Element
           code: 'VR',
           display: 'virtual',
         },
+        type: [appointment.appointmentType as CodeableConcept],
         serviceType: appointment.serviceType?.[0],
```
Comment (pamella) [inline examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx:93]: @rahul1 Please double-check if that's the correct mapping.

File: examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx [hunk 2]
```diff
@@ -41,21 +41,10 @@ export function AppointmentActions(props: AppointmentActionsProps): JSX.Element
...
       });
-      // If the appointment is cancelled, update the slot status to free
-      const slotId = resolveId(appointment.slot?.[0]);
-      if (newStatus === 'cancelled' && slotId) {
-        const slot = await medplum.readResource('Slot', slotId);
-        await medplum.updateResource({
-          ...slot,
-          status: 'free',
-        });
-      }
```
Comment (pamella) [inline examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx:58]: Simplify logic. It's best to keep the slot tied to the appointment for consistency, even if the appointment has been canceled.

File: examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx (hunk 1 above)
Comment (rahul1) [reply examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx:82]: Upon further research, I think this is not the right mapping.

in the create encounter modal, we should allow an (optional) input of the encounter type using the default bindings. https://hl7.org/fhir/R4/valueset-encounter-type.html

File: examples/medplum-scheduling-demo/src/components/CreateAppointment.tsx
Comment (rahul1) [inline examples/medplum-scheduling-demo/src/components/CreateAppointment.tsx:1]: One copy change: 

"'Which patient is the subject of this encounter?' -> 'Patient',

File: examples/medplum-scheduling-demo/data/core/encounter-types.json
Comment (rahul1) [inline examples/medplum-scheduling-demo/data/core/encounter-types.json:1]: let's rename this appointment-service-types

**Review by rahul1 — APPROVED** [review]
Minor comments

File: examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx (hunk 1 above)
Comment (pamella) [reply examples/medplum-scheduling-demo/src/components/AppointmentActions.tsx:82]: Currently, the encounter is created with a simple button click. I will add a modal so the users can input the encounter type before proceeding.

File: examples/medplum-scheduling-demo/README.md
Comment (pamella) [inline examples/medplum-scheduling-demo/README.md:1]: Some dev instructions. I'll review the entire content later on, as part of #4942

**Comment by pamella** [comment]
@rahul1 I've updated the PR to include the improvements we discussed during our sync today:
- Set up the calendar to scroll to the current time on the Schedule Page
- Update columns displayed in the Appointments list page
   - Add: status, appointment type, service type
   - Remove: last updated
 - Remove the _Mark Completed_ button
 - Add a modal to handle the Encounter creation flow
   - Add an info note about checking the Medplum Charting Demo
   - Add the type input

---

I'll tackle the cancelation flow adjustments in a separate PR to avoid clogging this one.

File: examples/medplum-scheduling-demo/src/components/CreateEncounter.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  title: 'Create Encounter',
+  status: 'active',
+  item: [
+    {
+      linkId: 'info-note',
+      type: 'display',
+      text:
+        "By submitting the form, the appointment status will be set to 'fulfilled' and " +
+        'an Encounter resource will be created, signaling the end of the Scheduling Demo flow. ' +
+        'To further explore the Encounter lifecycle, please visit the Medplum Charting Demo.',
```
Comment (pamella) [inline examples/medplum-scheduling-demo/src/components/CreateEncounter.tsx:102]: Let me know if there are adjustments to make in this copy.

Comment (rahul1) [reply examples/medplum-scheduling-demo/src/components/CreateEncounter.tsx:102]: By submitting the form, the appointment status will be set to 'fulfilled' and an Encounter resource will be created to capture details about the visit. To further explore the Encounter lifecycle, please visit the Medplum Charting Demo.

**Review by rahul1 — APPROVED** [review]

**Pull request merged**